src/modules/select/wallets/keepkey/index.ts

```diff
@@ -0,0 +1,402 @@
+  [ERROR_PAIRING]:
+    'There was an error pairing the device. Please disconnect and reconnect the device and try again.'
+}
+
+/* -= TYPES =- */
+
+type ErrorCode = 'busy' | 'pairing'
+
+type Balance = {
+  address: string
+  balance: string
+}
+
+interface KeepProviderOptions {
+  BigNumber: Helpers['BigNumber']
+  rpcUrl: string
+  resetWalletState: Helpers['resetWalletState']
+  networkId: number
+}
+
+function keepkey(
+  options: CommonWalletOptions & { rpcUrl: string; networkId: number }
+): WalletModule {
+  const { label, iconSrc, rpcUrl, networkId } = options
+
+  // Used to signal if the keep key could not be paired or if the keep key is already paired with another app
+  let installMessage: string
+  return {
+    name: label || WALLET_NAME,
+    iconSrc: iconSrc || keepKeyIcon,
+    wallet: async ({ BigNumber, resetWalletState }: Helpers) => {
+      const { provider, error } = await createKeepKeyProvider({
+        resetWalletState,
+        BigNumber,
+        rpcUrl,
+        networkId
+      })
+
+      installMessage = errorMessages[error as ErrorCode] || ''
+      return {
+        provider,
+        interface: !error
+          ? {
+              name: WALLET_NAME,
+              connect: provider.enable,
+              disconnect: provider.disconnect,
+              address: {
+                get: async () => provider.getPrimaryAddress()
+              },
+              network: {
+                get: async () => networkId
+              },
+              balance: {
+                get: async () => {
+                  const address = provider.getPrimaryAddress()
+                  return address && provider.getBalance(address)
+                }
+              }
+            }
+          : null
+      }
+    },
+    // The style tag here is used to hide the 'Open KeepKey button'
+    installMessage: () =>
+      installMessage
+        ? `<style>footer a > button { display: none !important }</style>
+        <p style="font-size: 0.889rem; font-family: inherit; margin: 0.889rem 0;">
+          ${installMessage}
+        </p>`
+        : '',
+    type: 'hardware',
+    mobile: false,
+    desktop: true
+  }
+}
+
+async function createKeepKeyProvider({
+  BigNumber,
+  rpcUrl,
+  resetWalletState,
+  networkId
+}: KeepProviderOptions): Promise<{ provider: any; error?: ErrorCode }> {
+  const { WebUSBKeepKeyAdapter } = await import(
+    '@shapeshiftoss/hdwallet-keepkey-webusb'
+  )
+
+  const {
+    Keyring,
+    Events,
+    bip32ToAddressNList,
+    HDWalletErrorType
+  } = await import('@shapeshiftoss/hdwallet-core')
+
+  const { default: createProvider } = await import('../providerEngine')
+  const { isValidPath } = await import('../hd-wallet')
+
+  // Initialize the adapter and pair the device
+  const keyring = new Keyring()
+  const keepKeyAdapter = WebUSBKeepKeyAdapter.useKeyring(keyring)
+  let keepKeyWallet: import('@shapeshiftoss/hdwallet-keepkey').KeepKeyHDWallet
```

Comment:
This is different from the async import method. Essentially it allows us to clearly indicate that we are importing only the type and prevents accidental inclusion of more than the type into the build. See [this post](https://davidea.st/articles/typescript-2-9-import-types) for more information.